upup/pkg/fi/cloudup/openstacktasks/floatingip.go

```diff
@@ -189,6 +189,21 @@ func (f *FloatingIP) RenderOpenstack(t *openstack.OpenstackAPITarget, a, e, chan
 			if err := e.Server.WaitForStatusActive(t); err != nil {
 				return fmt.Errorf("Failed to associate floating IP to instance %s", *e.Name)
 			}
+
+			// recheck is there floatingip already in port
+			// this can happen for instance when recreating bastion host
```

Comment:
how is the bastion host being recreated?

Reply:
if you run rolling-update for instance, it will delete machine (and then some another component will recreate it).

Reply:
the problem is that gophercloud will not return port id for floatingip (openstack api will return that but its not added to gophercloud). Otherwise we could check already before renderopenstack.

Reply:
I believe that this would be making a change for a component that doesn't exist yet in master.

Reply:
this could be managed by port task, we could just define do we need floatingip for port. Then porttask will create that if needed?

Reply:
I struggle with the compounded resources.  I like the idea of a task failing and being 1-to-1 with a resource.  I've even created resources that don't exist like poolassociation to make life easier here because I've had so much issue with doing multiple operations per task.

I'm going to test this by deleting my bastion server and re-running update.